src/codecs/decoding/format/native_json.rs

```diff
@@ -0,0 +1,51 @@
+use bytes::Bytes;
+use serde::{Deserialize, Serialize};
+use smallvec::{smallvec, SmallVec};
+use value::Kind;
+
+use super::Deserializer;
+use crate::{event::Event, schema};
+
+/// Config used to build a `NativeJsonDeserializer`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct NativeJsonDeserializerConfig;
+
+impl NativeJsonDeserializerConfig {
+    /// Build the `NativeJsonDeserializer` from this configuration.
+    pub const fn build(&self) -> NativeJsonDeserializer {
+        NativeJsonDeserializer
+    }
+
+    /// The schema produced by the deserializer.
+    pub fn schema_definition(&self) -> schema::Definition {
+        schema::Definition::empty().unknown_fields(Kind::json())
+    }
+}
+
+/// Deserializer that builds `Event`s from a byte frame containing JSON.
+#[derive(Debug, Clone, Default)]
+pub struct NativeJsonDeserializer;
+
+impl Deserializer for NativeJsonDeserializer {
+    fn parse(&self, bytes: Bytes) -> crate::Result<SmallVec<[Event; 1]>> {
+        // It's common to receive empty frames when parsing NDJSON, since it
+        // allows multiple empty newlines. We proceed without a warning here.
+        if bytes.is_empty() {
+            return Ok(smallvec![]);
+        }
+
+        // TODO: do we want to parse arrays?
```

Comment:
That's a good question, I'd lean to parsing arrays here. The only minor downside is that the roundtrip tests will be slightly more complicated when checking the events length. Is this TODO something we want to be resolved before going through with the PR?

Reply:
Yeah I leaned the same way. Since it isn't exercised in the roundtrip tests I just added a simpler standalone test for that specific behavior.